volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -0,0 +1,159 @@
+
+                    # MFT Flags determine the file type or dir
+                    if mft_record.Flags in mft_flags.choices.values():
+                        mft_flag = mft_flags.lookup(mft_record.Flags)
+                    else:
+                        mft_flag = hex(mft_record.Flags)
+
+                    # Standard Information Attribute
+                    if attr_header.AttrType == attr_types.choices.get('STANDARD_INFORMATION'):
+                        attr_data = self.context.object(si_object, offset = attr_data_offset, layer_name = layer.name)
+
+                        yield 0, (
+                            format_hints.Hex(attr_data_offset),
+                            mft_record.get_signature(),
+                            mft_record.RecordNumber,
+                            mft_record.LinkCount,
+                            mft_flag,
+                            renderers.NotApplicableValue(),
+                            attr_types.lookup(attr_header.AttrType),
+                            conversion.wintime_to_datetime(attr_data.CreationTime),
+                            conversion.wintime_to_datetime(attr_data.ModifiedTime),
+                            conversion.wintime_to_datetime(attr_data.UpdatedTime),
+                            conversion.wintime_to_datetime(attr_data.AccessedTime),
+                            renderers.NotApplicableValue(),
+                        )
+
+                    # File Name Attribute
+                    if attr_header.AttrType == attr_types.choices.get('FILE_NAME'):
+                        attr_data = self.context.object(fn_object, offset = attr_data_offset, layer_name = layer.name)
+                        file_name = attr_data.get_full_name()
+                        if attr_data.Flags in permission_flags.choices.values():
+                            permissions = permission_flags.lookup(attr_data.Flags)
+                        else:
+                            permissions = hex(attr_data.Flags)
+
+                        yield 1, (format_hints.Hex(attr_data_offset), mft_record.get_signature(),
+                                  mft_record.RecordNumber, mft_record.LinkCount, mft_flag, permissions,
+                                  attr_types.lookup(attr_header.AttrType),
+                                  conversion.wintime_to_datetime(attr_data.CreationTime),
+                                  conversion.wintime_to_datetime(attr_data.ModifiedTime),
+                                  conversion.wintime_to_datetime(attr_data.UpdatedTime),
+                                  conversion.wintime_to_datetime(attr_data.AccessedTime), file_name)
+
+                    # Update the base offset to point to the next attribute
+                    attr_base_offset += attr_header.Length
+
+            except exceptions.PagedInvalidAddressException:
+                pass
+
+    def generate_timeline(self):
+        print("MFT Timeline")
```

Comment:
Debugging statements should be done with `vollog` (I've had these sneak in occasionally, but since we've spotted it, best to remove it please).  5;P

Reply:
Silly mistake not to check 😁